vortex-array/src/compute/conformance/binary_numeric.rs

```diff
@@ -19,11 +41,50 @@ fn to_vec_of_scalar(array: &dyn Array) -> Vec<Scalar> {
         .vortex_unwrap()
 }
 
-pub fn test_binary_numeric_conformance<T: NativePType + Num + Copy>(array: ArrayRef)
+/// Tests binary numeric operations for conformance across array encodings.
+///
+/// # Type Parameters
+///
+/// * `T` - The native numeric type (e.g., i32, f64) that the array contains
+///
+/// # Arguments
+///
+/// * `array` - The array to test, which should contain numeric values of type `T`
+///
+/// # Test Details
+///
+/// This function:
+/// 1. Canonicalizes the input array to primitive form to get expected values
+/// 2. Tests all binary numeric operators against a constant value of 1
+/// 3. Verifies results match the expected primitive array computation
+/// 4. Tests both array-operator-scalar and scalar-operator-array forms
+/// 5. Gracefully skips operations that would cause overflow/underflow
+///
+/// # Panics
+///
+/// Panics if:
+/// - The array cannot be converted to primitive form
+/// - Results don't match expected values (for operations that don't overflow)
+fn test_binary_numeric_conformance<T: NativePType + Num + Copy>(array: ArrayRef)
 where
     Scalar: From<T>,
 {
-    let canonicalized_array = array.to_primitive().vortex_unwrap();
+    // First test with the standard scalar value of 1
+    test_standard_binary_numeric::<T>(array.clone());
+
+    // Then test edge cases if we have enough data
+    if array.len() >= 5 {
+        test_binary_numeric_edge_cases(array);
```

Comment:
I don't think any of the tests in here rely on the array being greater than or equal to 5.